src/rgw/rgw_asio_frontend.cc

```diff
@@ -134,6 +136,10 @@ void handle_connection(boost::asio::io_context& context,
                        boost::system::error_code& ec,
                        spawn::yield_context yield)
 {
+
+  std::call_once(tracerInit,[](){
```

Comment:
In rgw_main and common inti are fired only once during startup so if the jaeger server is off during startup of rgw then it wont be initialized and we wont get the spans even if `WITH_JAEGER` is defined, but if I place my jaeger initialization inside client code then it will initialized everytime a request comes and there wont be any risk of missing jaeger initialization when jaeger server is off

Reply:
ok, that makes sense. 2 issues, though:
* what is someone want to start tracing from somewhere else in the code? won't they need to share the same "call_once" flag?
* for retry to work, you need to get of of the init function using an exception - is this implemented?

Reply:
"what is someone want to start tracing from somewhere else in the code? won't they need to share the same "call_once" flag?" yeah correct I need to make my once_flag extern.
"for retry to work, you need to get of of the init function using an exception - is this implemented?" there is no exception when a connection with jaeger server fails to happen it will just pass with a error message.

Reply:
> "for retry to work, you need to get of of the init function using an exception - is this implemented?" there is no exception when a connection with jaeger server fails to happen it will just pass with a error message.

so, retry won't work, because `call_once()` would be successful. if you want retry to work, you should wrap your call with check to the return code, and throw an error if it wasn't successful, so that `call_once()` would fail, and be invoked again on the next trace.

Reply:
@suab321321 have you considered checking is tracer is initialized using an 
``` if ( ! tracer ){
create new tracer}
``` 
condition?